src/components/BookList.js

```diff
@@ -1,21 +1,39 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { removeBook } from '../redux/books/booksSlice';
+import { fetchBooks, removeBook } from '../redux/books/booksSlice';
 import Book from './Book';
+import Form from './Form';
 
 function BookList() {
-  const books = useSelector((state) => state.books);
+  const { books, isLoading, error } = useSelector((state) => state.books);
   const dispatch = useDispatch();
 
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
+
   const handleDelete = (id) => {
-    // Logic to delete the book
-    dispatch(removeBook({ id }));
+    dispatch(removeBook(id));
   };
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (error) {
+    return <p>Could not fetch data.</p>;
+  }
+
+  if (!books) {
+    return null;
+  }
```

Comment:
- [ ] Please make sure that your app is working as expected, :pray: Add and remove books should work as expected but currently, only your app navigation bar is rendering, As you can see below the form for adding and removing books rendered the first time and then disappear so make sure that you fix this :point_down: 
  
  [Screencast from 07-18-2023 08:57:09 PM.webm](https://github.com/Zohra-Neda/Bookstore/assets/73854999/1c34b39e-548e-4cc0-88b7-656ad7f944e1)